lib/screens/v2/widget_instance/reconstructed_alert.ex

```diff
@@ -603,13 +624,21 @@ defmodule Screens.V2.WidgetInstance.ReconstructedAlert do
   def serialize(%__MODULE__{} = t) do
     case BaseAlert.location(t) do
       :inside ->
-        serialize_inside_alert(t)
+        t |> serialize_inside_alert() |> Map.put(:region, :inside)
 
       location when location in [:boundary_upstream, :boundary_downstream] ->
-        serialize_boundary_alert(t)
+        t |> serialize_boundary_alert() |> Map.put(:region, :boundary)
 
       location when location in [:downstream, :upstream] ->
-        serialize_outside_alert(t)
+        t |> serialize_outside_alert() |> Map.put(:region, :outside)
+    end
+  end
+
+  def audio_sort_key(%__MODULE__{} = t) do
+    case serialize(t) do
+      %{urgent: true} -> [2]
+      %{effect: effect} when effect in [:moderate_delay, :delay] -> [2, 2]
+      _ -> [2, 1]
```

Comment:
So that is correct when it comes to the visual version of this function `priority/1`. In that function, 1 is reserved for (most of the time) takeovers. In audio, I think we need to move away from that because each widget (or in the case of alerts, widget group) needs to be read in a particular order. So for this widget, I am giving alerts a default order of 2. Any number after that will determine the priority of the alert in relation to other alerts. Urgent alerts having [2, 1] will ensure that they are always in the front of the group.

Reply:
I see. Would we not want `%{urgent: true} -> [2, 1]` then and the catch-all be `_ -> [2, 1]`?

Reply:
Well in Elixir, `[2]` is less than `[2, 1]`. Array comparison looks at each item individually and moves to the next item for tiebreakers. If there is no next item, that array wins out. This would mean urgent alerts will always have a lesser sort key than other alerts because it has no tiebreaker.

Reply:
Ahh okay I see.